compiler/src/iree/compiler/Dialect/Util/Analysis/Explorer.cpp

```diff
@@ -112,6 +118,7 @@ void Explorer::initializeGlobalInfos() {
     } else {
       globalInfo.uses.push_back(use.getUser());
     }
+    orderedGlobalInfos.push_back(&globalInfo);
```

Comment:
W.R.T the deterministic processing of globals I did this, but I don't think it's good that it's tracking every global info twice for all other users of Explorer that don't care about this.